### PR TITLE
GH-212 - disable CallInfo analysis for Arg.AnyType matchers

### LINKED ISSUE #212: Disable CallInfo analysis when Arg.AnyType used for argument matching
https://github.com/nsubstitute/NSubstitute/pull/715#issuecomment-1708934821
https://github.com/nsubstitute/NSubstitute/pull/715#issuecomment-1712043274

### PR DESCRIPTION
**Pull request by tpodolak**
Closes #212

### CODE REVIEW

File: .config/dotnet-tools.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (tpodolak) [inline .config/dotnet-tools.json:1]: Had to upgrade cake as current appveyor win images dont have required .netcore version (2.x.x)
https://ci.appveyor.com/project/NSubstitute/nsubstitute-analyzers/builds/48719348/job/d4iwb70xjucs5cag

Comment (alexandrnikitin) [reply .config/dotnet-tools.json:1]: @tpodolak We've migrated the NSub repo to GitHub actions recently https://github.com/nsubstitute/NSubstitute/blob/main/.github/workflows/build_and_test.yml
I can help with that if you want.

**Review by dtchepak — APPROVED** [review]
Thanks @tpodolak :)

File: .config/dotnet-tools.json (hunk 1 above)
Comment (tpodolak) [reply .config/dotnet-tools.json:1]: Thanks @alexandrnikitin. I've added task to migrate analyzers as well https://github.com/nsubstitute/NSubstitute.Analyzers/issues/216

**Pull request merged**